week4/login/src/components/MyPage/Header.tsx

```diff
@@ -0,0 +1,33 @@
+/** @jsxImportSource @emotion/react */
+import { useNavigate } from "react-router-dom";
+import { headerStyle, leftStyle, rightStyle } from "./Header.style";
+
+interface HeaderType {
+  selected: string;
+  onChange: (value: string) => void;
+}
+
+const Header: React.FC<HeaderType> = ({ onChange }) => {
```

Comment:
React.FC를 사용하셨는데, 요즘 FC는 지양하는 편입니다!

제네릭을 지원하지 않고, children이 기본적으로 들어가있기 때문입니다!!
저희가 타입스크립트를 사용하는 이유는 타입을 정확히 지정해줘서 코드의 안정성을 향상시키기 위함인데요, FC를 사용하면 children을 타입 지정하지 않고 전달이 가능해서 안정성을 떨어뜨립니다!! 
또, 현재 이 코드에서는 children을 사용하지 않고 있기 때문에 FC를 사용하지 않는 것이 더 적합합니다!

```suggestion

const Header = ({ onChange }: HeaderType) => {
```

이렇게 작성해서 더 간결하고 명확하게 타입을 정의해보는 것은 어떨까요?

함께 읽어보면 좋을 것 같아서 아티클도 첨부합니다!!
https://f-lab.kr/insight/react-fc-type-usage-20241102